packages/admin-ui/src/combobox/combobox.state.ts

```diff
@@ -33,24 +76,58 @@ export function useComboboxState(props: ComboboxStateProps = {}) {
     }
 
     return 'ready'
-  }, [loading, error, state.matches, state.value])
+  }, [loading, error, matches, state.value])
 
   return {
+    ...state,
     deferredValue,
     setError,
     setLoading,
     status,
-    ...state,
+    getOptionValue,
+    renderOption,
+    setSelectedItem,
+    selectedItem,
+    setMatches,
+    matches,
   }
 }
 
 type Status = 'loading' | 'error' | 'empty-search' | 'no-result' | 'ready'
 
-export type ComboboxStateProps = Pick<
+export type ComboboxStateProps<T> = Pick<
   AriakitComboboxStateProps,
-  'list' | 'virtualFocus'
+  'virtualFocus'
 > & {
+  /** Initial list of values that will be filtered on search, should be set unless the input will be controlled */
+  list?: T[]
+  /** Function for transforming item shape into a string value, no need to use it on string[] lists */
+  getOptionValue?: (item: T) => string
+  /** Function for transforming item shape into renderable node, no need to use it on string[] lists */
+  renderOption?: (item: T) => ReactNode
+  /** Debounce interval */
   timeoutMs?: number
 }
 
-export type ComboboxState = ReturnType<typeof useComboboxState>
+export type ComboboxState<T> = Omit<AriakitComboboxState, 'matches'> & {
```

Comment:
What's wrong with the `ReturnType`?

Reply:
I needed to use the generic type on `matches: T[]`

Reply:
Makes sense